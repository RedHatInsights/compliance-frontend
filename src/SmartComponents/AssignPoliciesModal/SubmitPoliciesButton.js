import React from 'react';
import { useMutation } from '@apollo/react-hooks';
import { Button } from '@patternfly/react-core';
import propTypes from 'prop-types';
import { ASSOCIATE_PROFILES_TO_SYSTEM } from '../../Utilities/graphql/mutations';
import { addNotification } from '@redhat-cloud-services/frontend-components-notifications';
import { dispatchAction } from '../../Utilities/Dispatcher';
import { reset } from 'redux-form';

export const completedMessage = (system) => {
    const profiles = system.profiles.map((profile) => profile.name);
    let message;
    if (profiles.length > 0) {
        message = `Associated ${profiles} to ${system.name}`;
    } else {
        message = `${system.name} is now not associated to any policy`;
    }

    return message;
};

export const SubmitPoliciesButton = ({ policyIds, systemId, toggle, dispatch }) => {
    const [associateProfilesToSystem] = useMutation(ASSOCIATE_PROFILES_TO_SYSTEM, {
        onCompleted: (data) => {
            dispatchAction(addNotification({
                variant: 'success',
                title: completedMessage(data.associateProfiles.system)
            }));
            toggle();
            dispatch(reset('assignPolicies'));
        },
        onError: (error) => {
            dispatchAction(addNotification({
                variant: 'danger',
                title: 'Error associating policies',
                description: error.message
            }));
            toggle();
            dispatch(reset('assignPolicies'));
        }
    });

    return <Button type='submit' aria-label='save'
        onClick={() => associateProfilesToSystem({ variables: { input: { profileIds: policyIds, id: systemId } } })}
        variant='primary'>Save</Button>;
};

SubmitPoliciesButton.propTypes = {
    policyIds: propTypes.arrayOf(propTypes.string).isRequired,
    systemId: propTypes.string.isRequired,
    toggle: propTypes.func.isRequired,
    dispatch: propTypes.func
};

SubmitPoliciesButton.defaultProps = {
    policyIds: []
};
