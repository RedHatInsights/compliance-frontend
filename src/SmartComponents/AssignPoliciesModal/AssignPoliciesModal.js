import React from 'react';
import propTypes from 'prop-types';
import { CheckboxGroup, ErrorPage } from 'PresentationalComponents';
import {
    Modal,
    Text,
    TextContent,
    Button
} from '@patternfly/react-core';
import { useQuery } from '@apollo/react-hooks';
import { SubmitPoliciesButton } from './SubmitPoliciesButton';
import gql from 'graphql-tag';
import { reset, reduxForm, formValueSelector } from 'redux-form';
import { connect } from 'react-redux';
import { compose } from 'redux';

const QUERY = gql`
query systemWithProfiles($search: String!){
    systems(search: $search) {
        nodes {
            id
            name
            profiles {
                id
                name
            }
        }
    }
    profiles {
        nodes {
            id
            name
        }
    }
}
`;

const isSystemAssignedToProfile = (system, id) => (
    system.profiles.map(profile => profile.id).includes(id)
);

const AssignPoliciesModal = ({ isModalOpen, toggle, fqdn, id, selectedPolicyIds, dispatch }) => {
    // Display all policies with the same OS as host
    const { data, error, loading } = useQuery(QUERY, { variables: { search: 'id=' + id } });

    if (error) { return <ErrorPage error={error}/>; }

    if (loading) { return <React.Fragment/>; }

    const system = data.systems.nodes.length === 1 && data.systems.nodes[0];
    const options = data.profiles.nodes.map((policy) => (
        {
            label: policy.name,
            value: policy.id,
            defaultChecked: system && isSystemAssignedToProfile(system, policy.id)
        }
    ));

    return (
        <Modal
            isSmall
            title={`Edit policies for ${fqdn}`}
            isOpen={isModalOpen}
            isFooterLeftAligned
            onClose={() => { dispatch(reset('assignPolicies')); toggle(true); }}
            actions={[
                <SubmitPoliciesButton key='save'
                    aria-label='save'
                    toggle={toggle}
                    systemId={id}
                    dispatch={dispatch}
                    policyIds={selectedPolicyIds}
                    variant='primary' />,
                <Button key='cancel' aria-label='cancel' variant='secondary'
                    onClick={() => { dispatch(reset('assignPolicies')); toggle(true); }}>
                    Cancel
                </Button>
            ]}
        >
            <TextContent>
                <Text>
                    Choose which policies {fqdn} should be a part of. Note: Only policies for this
                    host operating system are shown.
                </Text>
            </TextContent>
            <br/>
            <TextContent>
                <Text>
                    <CheckboxGroup name='policies' options={options} />
                </Text>
            </TextContent>
        </Modal>
    );
};

AssignPoliciesModal.propTypes = {
    isModalOpen: propTypes.bool,
    toggle: propTypes.func,
    id: propTypes.string,
    fqdn: propTypes.string,
    selectedPolicyIds: propTypes.arrayOf(propTypes.string),
    dispatch: propTypes.func
};

const selector = formValueSelector('assignPolicies');

export default compose(
    connect(
        state => ({
            selectedPolicyIds: selector(state, 'policies')
        })
    ),
    reduxForm({
        form: 'assignPolicies'
    })
)(AssignPoliciesModal);
