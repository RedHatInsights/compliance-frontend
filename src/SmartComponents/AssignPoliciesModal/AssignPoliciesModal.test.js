import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { act } from 'react-dom/test-utils';
import { useQuery, useMutation } from '@apollo/react-hooks';

const waitForComponentToPaint = async (wrapper: any) => {
    await act(async () => {
        await new Promise(resolve => setTimeout(resolve, 0));
        wrapper.update();
    });
};

const mockStore = configureStore();
jest.mock('@apollo/react-hooks');
jest.mock('Utilities/InventoryApi', () => (
    {
        /* eslint-disable camelcase */
        getSystemProfile: () => Promise.resolve({ results: [{ system_profile: { os_release: '8.1' } }] })
        /* eslint-enable camelcase */
    }
));
jest.mock('../CreatePolicy/EditPolicyRules', () => {
    return <p>Rules table</p>;
});
jest.mock('@redhat-cloud-services/frontend-components-inventory-compliance', () => {
    const ComplianceRemediationButton = () => <button>Remediations</button>;
    return ComplianceRemediationButton;
});

import AssignPoliciesModal from './AssignPoliciesModal.js';

describe('AssignPoliciesModal', () => {
    const defaultProps = {
        fqdn: 'myhost',
        id: '1',
        toggle: jest.fn()
    };

    let store;
    beforeEach(() => {
        store = mockStore({ form: { policyForm: {
            values: { policies: ['1', '2'] }
        } } });

        useMutation.mockImplementation((query, options) => {
            return [function() {
                options.onCompleted();
            }];
        });
    });

    it('expect to render an open modal without error', async () => {
        useQuery.mockImplementation(() => ({
            data: {
                systems: {
                    nodes: [
                        {
                            id: '1',
                            fqdn: 'myhost',
                            profiles: [
                                {
                                    id: '1',
                                    name: 'profile1',
                                    majorOsVersion: '8'
                                }
                            ]
                        }
                    ]
                },
                profiles: {
                    nodes: [{
                        id: '1',
                        name: 'profile1',
                        majorOsVersion: '8'
                    }, {
                        id: '2',
                        name: 'profile2',
                        majorOsVersion: '7'
                    }]
                }
            }, error: false, loading: false }));

        const component = mount(
            <Provider store={store}>
                <AssignPoliciesModal { ...defaultProps } isModalOpen={ true } />
            </Provider>
        );
        await waitForComponentToPaint(component);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('expect not to render anything for a closed modal', async () => {
        useQuery.mockImplementation(() => ({
            data: {
                systems: {
                    nodes: [
                        {
                            id: '1',
                            fqdn: 'myhost',
                            profiles: [
                                {
                                    id: '1',
                                    name: 'profile1',
                                    majorOsVersion: '8'
                                }
                            ]
                        }
                    ]
                },
                profiles: {
                    nodes: [{
                        id: '1',
                        name: 'profile1',
                        majorOsVersion: '8'
                    }]
                }
            }, error: false, loading: false }));

        const component = mount(
            <Provider store={store}>
                <AssignPoliciesModal { ...defaultProps } isModalOpen={ false } />
            </Provider>
        );
        await waitForComponentToPaint(component);
        expect(toJson(component)).toMatchSnapshot();
    });

    it('renders empty state if no OS is matching the system', async () => {
        useQuery.mockImplementation(() => ({
            data: {
                systems: {
                    nodes: [
                        {
                            id: '1',
                            fqdn: 'myhost'
                        }
                    ]
                },
                profiles: {
                    nodes: [{
                        id: '1',
                        name: 'profile1',
                        majorOsVersion: '7'
                    }, {
                        id: '2',
                        name: 'profile2',
                        majorOsVersion: '7'
                    }]
                }
            }, error: false, loading: false }));

        const component = mount(
            <Provider store={store}>
                <AssignPoliciesModal { ...defaultProps } isModalOpen={ open } />
            </Provider>
        );
        await waitForComponentToPaint(component);
        expect(toJson(component)).toMatchSnapshot();
    });
});
