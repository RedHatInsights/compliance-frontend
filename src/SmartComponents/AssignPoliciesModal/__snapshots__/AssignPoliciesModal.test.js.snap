// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignPoliciesModal expect not to render anything for a closed modal 1`] = `
<Modal
  actions={
    Array [
      <SubmitPoliciesButton
        aria-label="save"
        dispatch={[Function]}
        policyIds={Array []}
        systemId="1"
        toggle={[MockFunction]}
        variant="primary"
      />,
      <Button
        aria-label="cancel"
        onClick={[Function]}
        variant="secondary"
      >
        Cancel
      </Button>,
    ]
  }
  appendTo={
    <body
      class=""
    >
      <div>
        <div
          class="pf-c-backdrop"
        >
          <div
            class="pf-l-bullseye"
          >
            <div
              aria-describedby="pf-modal-part-2"
              aria-labelledby="pf-modal-part-1"
              aria-modal="true"
              class="pf-c-modal-box pf-m-sm"
              data-ouia-component-id="0"
              data-ouia-component-type="PF4/ModalContent"
              data-ouia-safe="true"
              id="pf-modal-part-0"
              role="dialog"
            >
              <button
                aria-disabled="false"
                aria-label="Close"
                class="pf-c-button pf-m-plain"
                data-ouia-component-id="1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 352 512"
                  width="1em"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    transform=""
                  />
                </svg>
              </button>
              <header
                class="pf-c-modal-box__header"
              >
                <h1
                  class="pf-c-modal-box__title pf-c-modal-box__title"
                  id="pf-modal-part-1"
                >
                  Edit policies for myhost
                </h1>
              </header>
              <div
                class="pf-c-modal-box__body"
                id="pf-modal-part-2"
              >
                <div
                  class="pf-c-content"
                >
                  <p
                    class=""
                    data-pf-content="true"
                  >
                    Choose which policies 
                    myhost
                     should be a part of. Note: Only policies for this host operating system are shown.
                  </p>
                </div>
                <br />
                <div
                  class="pf-c-content"
                >
                  <p
                    class=""
                    data-pf-content="true"
                  >
                    <div
                      class="pf-c-check"
                    >
                      <input
                        aria-invalid="false"
                        checked=""
                        class="pf-c-check__input"
                        id="1"
                        name="policies[0]"
                        type="checkbox"
                        value="1"
                      />
                      <label
                        class="pf-c-check__label"
                        for="1"
                      >
                        profile1
                      </label>
                    </div>
                  </p>
                </div>
              </div>
              <footer
                class="pf-c-modal-box__footer"
              >
                <button
                  aria-disabled="false"
                  aria-label="save"
                  class="pf-c-button pf-m-primary"
                  data-ouia-component-id="2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="submit"
                >
                  Save
                </button>
                <button
                  aria-disabled="false"
                  aria-label="cancel"
                  class="pf-c-button pf-m-secondary"
                  data-ouia-component-id="3"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Cancel
                </button>
              </footer>
            </div>
          </div>
        </div>
      </div>
      <div />
    </body>
  }
  aria-describedby=""
  aria-label=""
  aria-labelledby=""
  className=""
  hasNoBodyWrapper={false}
  isFooterLeftAligned={true}
  isOpen={false}
  onClose={[Function]}
  ouiaSafe={true}
  showClose={true}
  title="Edit policies for myhost"
  variant="small"
/>
`;

exports[`AssignPoliciesModal expect to render an open modal without error 1`] = `
<Modal
  actions={
    Array [
      <SubmitPoliciesButton
        aria-label="save"
        dispatch={[Function]}
        policyIds={Array []}
        systemId="1"
        toggle={[MockFunction]}
        variant="primary"
      />,
      <Button
        aria-label="cancel"
        onClick={[Function]}
        variant="secondary"
      >
        Cancel
      </Button>,
    ]
  }
  appendTo={
    <body
      class="pf-c-backdrop__open"
    >
      <div>
        <div
          class="pf-c-backdrop"
        >
          <div
            class="pf-l-bullseye"
          >
            <div
              aria-describedby="pf-modal-part-2"
              aria-labelledby="pf-modal-part-1"
              aria-modal="true"
              class="pf-c-modal-box pf-m-sm"
              data-ouia-component-id="0"
              data-ouia-component-type="PF4/ModalContent"
              data-ouia-safe="true"
              id="pf-modal-part-0"
              role="dialog"
            >
              <button
                aria-disabled="false"
                aria-label="Close"
                class="pf-c-button pf-m-plain"
                data-ouia-component-id="1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 352 512"
                  width="1em"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    transform=""
                  />
                </svg>
              </button>
              <header
                class="pf-c-modal-box__header"
              >
                <h1
                  class="pf-c-modal-box__title pf-c-modal-box__title"
                  id="pf-modal-part-1"
                >
                  Edit policies for myhost
                </h1>
              </header>
              <div
                class="pf-c-modal-box__body"
                id="pf-modal-part-2"
              >
                <div
                  class="pf-c-content"
                >
                  <p
                    class=""
                    data-pf-content="true"
                  >
                    Choose which policies 
                    myhost
                     should be a part of. Note: Only policies for this host operating system are shown.
                  </p>
                </div>
                <br />
                <div
                  class="pf-c-content"
                >
                  <p
                    class=""
                    data-pf-content="true"
                  >
                    <div
                      class="pf-c-check"
                    >
                      <input
                        aria-invalid="false"
                        checked=""
                        class="pf-c-check__input"
                        id="1"
                        name="policies[0]"
                        type="checkbox"
                        value="1"
                      />
                      <label
                        class="pf-c-check__label"
                        for="1"
                      >
                        profile1
                      </label>
                    </div>
                  </p>
                </div>
              </div>
              <footer
                class="pf-c-modal-box__footer"
              >
                <button
                  aria-disabled="false"
                  aria-label="save"
                  class="pf-c-button pf-m-primary"
                  data-ouia-component-id="2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="submit"
                >
                  Save
                </button>
                <button
                  aria-disabled="false"
                  aria-label="cancel"
                  class="pf-c-button pf-m-secondary"
                  data-ouia-component-id="3"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Cancel
                </button>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </body>
  }
  aria-describedby=""
  aria-label=""
  aria-labelledby=""
  className=""
  hasNoBodyWrapper={false}
  isFooterLeftAligned={true}
  isOpen={true}
  onClose={[Function]}
  ouiaSafe={true}
  showClose={true}
  title="Edit policies for myhost"
  variant="small"
/>
`;

exports[`AssignPoliciesModal renders empty state if no OS is matching the system 1`] = `
<Modal
  actions={
    Array [
      <SubmitPoliciesButton
        aria-label="save"
        dispatch={[Function]}
        policyIds={Array []}
        systemId="1"
        toggle={[MockFunction]}
        variant="primary"
      />,
      <Button
        aria-label="cancel"
        onClick={[Function]}
        variant="secondary"
      >
        Cancel
      </Button>,
    ]
  }
  appendTo={
    <body
      class="pf-c-backdrop__open"
    >
      <div
        aria-hidden="true"
      >
        <div
          class="pf-c-backdrop"
        >
          <div
            class="pf-l-bullseye"
          >
            <div
              aria-describedby="pf-modal-part-2"
              aria-labelledby="pf-modal-part-1"
              aria-modal="true"
              class="pf-c-modal-box pf-m-sm"
              data-ouia-component-id="0"
              data-ouia-component-type="PF4/ModalContent"
              data-ouia-safe="true"
              id="pf-modal-part-0"
              role="dialog"
            >
              <button
                aria-disabled="false"
                aria-label="Close"
                class="pf-c-button pf-m-plain"
                data-ouia-component-id="1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 352 512"
                  width="1em"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    transform=""
                  />
                </svg>
              </button>
              <header
                class="pf-c-modal-box__header"
              >
                <h1
                  class="pf-c-modal-box__title pf-c-modal-box__title"
                  id="pf-modal-part-1"
                >
                  Edit policies for myhost
                </h1>
              </header>
              <div
                class="pf-c-modal-box__body"
                id="pf-modal-part-2"
              >
                <div
                  class="pf-c-content"
                >
                  <p
                    class=""
                    data-pf-content="true"
                  >
                    Choose which policies 
                    myhost
                     should be a part of. Note: Only policies for this host operating system are shown.
                  </p>
                </div>
                <br />
                <div
                  class="pf-c-content"
                >
                  <p
                    class=""
                    data-pf-content="true"
                  >
                    <div
                      class="pf-c-check"
                    >
                      <input
                        aria-invalid="false"
                        checked=""
                        class="pf-c-check__input"
                        id="1"
                        name="policies[0]"
                        type="checkbox"
                        value="1"
                      />
                      <label
                        class="pf-c-check__label"
                        for="1"
                      >
                        profile1
                      </label>
                    </div>
                  </p>
                </div>
              </div>
              <footer
                class="pf-c-modal-box__footer"
              >
                <button
                  aria-disabled="false"
                  aria-label="save"
                  class="pf-c-button pf-m-primary"
                  data-ouia-component-id="2"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="submit"
                >
                  Save
                </button>
                <button
                  aria-disabled="false"
                  aria-label="cancel"
                  class="pf-c-button pf-m-secondary"
                  data-ouia-component-id="3"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Cancel
                </button>
              </footer>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
      />
      <div>
        <div
          class="pf-c-backdrop"
        >
          <div
            class="pf-l-bullseye"
          >
            <div
              aria-describedby="pf-modal-part-4"
              aria-labelledby="pf-modal-part-3"
              aria-modal="true"
              class="pf-c-modal-box pf-m-sm"
              data-ouia-component-id="4"
              data-ouia-component-type="PF4/ModalContent"
              data-ouia-safe="true"
              id="pf-modal-part-2"
              role="dialog"
            >
              <button
                aria-disabled="false"
                aria-label="Close"
                class="pf-c-button pf-m-plain"
                data-ouia-component-id="5"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 352 512"
                  width="1em"
                >
                  <path
                    d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    transform=""
                  />
                </svg>
              </button>
              <header
                class="pf-c-modal-box__header"
              >
                <h1
                  class="pf-c-modal-box__title pf-c-modal-box__title"
                  id="pf-modal-part-3"
                >
                  Edit policies for myhost
                </h1>
              </header>
              <div
                class="pf-c-modal-box__body"
                id="pf-modal-part-4"
              >
                <div
                  class="pf-c-content"
                >
                  <p
                    class=""
                    data-pf-content="true"
                  >
                    Choose which policies 
                    myhost
                     should be a part of. Note: Only policies for this host operating system are shown.
                  </p>
                </div>
                <br />
                <div
                  class="pf-c-content"
                >
                  <p
                    class=""
                    data-pf-content="true"
                  >
                    No eligible policies for this system.
                  </p>
                </div>
              </div>
              <footer
                class="pf-c-modal-box__footer"
              >
                <button
                  aria-disabled="false"
                  aria-label="save"
                  class="pf-c-button pf-m-primary"
                  data-ouia-component-id="6"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="submit"
                >
                  Save
                </button>
                <button
                  aria-disabled="false"
                  aria-label="cancel"
                  class="pf-c-button pf-m-secondary"
                  data-ouia-component-id="7"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Cancel
                </button>
              </footer>
            </div>
          </div>
        </div>
      </div>
    </body>
  }
  aria-describedby=""
  aria-label=""
  aria-labelledby=""
  className=""
  hasNoBodyWrapper={false}
  isFooterLeftAligned={true}
  isOpen={[Function]}
  onClose={[Function]}
  ouiaSafe={true}
  showClose={true}
  title="Edit policies for myhost"
  variant="small"
/>
`;
